Error installing zarith-xen 1.3

$ opam reinstall zarith-xen
    The following actions will be performed:
      ↻  recompile zarith-xen     1.3    
      ↻  recompile nocrypto       0.4.0           [uses zarith-xen]
      ↻  recompile x509           0.3.1           [uses nocrypto]
      ↻  recompile tls            0.5.0           [uses nocrypto]
      ↻  recompile conduit        0.8.4*          [uses tls]
      ↻  recompile mirage-conduit 2.2.0           [uses conduit]
      ↻  recompile cohttp         0.17.2*         [uses conduit]
      ↻  recompile mirage-http    2.3.0           [uses cohttp, mirage-conduit]
      ↻  recompile irmin          0.9.4*          [uses nocrypto]

[ ...lots of warnings... ]

    - /usr/bin/ld: /home/mirage/.opam/profile/lib/pkgconfig/../gmp-xen/libgmp-xen.a(lt4-clear.o): relocation R_ARM_MOVW_ABS_NC against `__gmp_free_func' can not be used when making a shared object; recompile with -fPIC
    - /home/mirage/.opam/profile/lib/pkgconfig/../gmp-xen/libgmp-xen.a: error adding symbols: Bad value
    - collect2: error: ld returned 1 exit status
    - ocamlopt   -c q.ml
    - ocamlopt   -c big_int_Z.ml
    - ocamlmklib -failsafe -o zarith z.cmx q.cmx big_int_Z.cmx -L/home/mirage/.opam/profile/lib/pkgconfig/../gmp-xen -lgmp-xen   -lgmp
    - ocamlopt -shared -o zarith.cmxs -I . zarith.cmxa
    - + cp libzarith.a /home/mirage/.opam/profile/lib/zarith/libzarith-xen.a
    - + mv /home/mirage/.opam/profile/lib/zarith/META /home/mirage/.opam/profile/lib/zarith/META.old
    - mv: cannot stat ‘/home/mirage/.opam/profile/lib/zarith/META’: No such file or directory
    [ERROR] The compilation of zarith-xen failed at "./mirage-install.sh".
    [zarith-xen: rm mv] Command started
    + rm "-f" "/home/mirage/.opam/profile/lib/zarith/libzarith-xen.a" "mv" "/home/mirage/.opam/profile/lib/zarith/META.old" "/home/mirage/.opam/profile/lib/zarith/META" (CWD=/home/mirage/.opam/profile/build/zarith-xen.1.3)

    #=== ERROR while installing zarith-xen.1.3 ====================================#
    # opam-version 1.2.1
    # os           linux
    # command      ./mirage-install.sh
    # path         /home/mirage/.opam/profile/build/zarith-xen.1.3
    # compiler     system (4.01.0)
    # exit-code    1
    # env-file     /home/mirage/.opam/profile/build/zarith-xen.1.3/zarith-xen-8074-487c34.env
    # stdout-file  /home/mirage/.opam/profile/build/zarith-xen.1.3/zarith-xen-8074-487c34.out
    # stderr-file  /home/mirage/.opam/profile/build/zarith-xen.1.3/zarith-xen-8074-487c34.err
    ### stdout ###
    # [...]
    # ocamlc   -c big_int_Z.ml
    # ocamlmklib -failsafe -o zarith z.cmo q.cmo big_int_Z.cmo -L/home/mirage/.opam/profile/lib/pkgconfig/../gmp-xen -lgmp-xen   -lgmp
    # gcc -DZ_OCAML_HASH -DZ_OCAML_COMPARE_EXT -DHAS_GMP     -c -o caml_z_arm.o caml_z_arm.S
    # ocamlc -ccopt "-I/usr/lib/ocaml  -DZ_OCAML_HASH -DZ_OCAML_COMPARE_EXT -DHAS_GMP  -O3 -Wall -Wextra -isystem/home/mirage/.opam/profile/lib/pkgconfig/../../include/minios-xen -isystem/home/mirage/.opam/profile/lib/pkgconfig/../../include/minios-xen/posix -march=armv7-a -marm -fms-extensions -D__arm__ -DXEN_HAVE_PV_GUEST_ENTRY -nostdinc -isystem /usr/lib/gcc/arm-linux-gnueabihf/4.8/include -U__linux__ -U__FreeBSD__ -U__sun__ -U__linux -D__MINIOS__ -I/home/mirage/.opam/profile/lib/pkgconfig/../gmp-xen/include -I/home/mirage/.opam/profile/lib/pkgconfig/../../include/mirage-xen-posix/include -I/home/mirage/.opam/profile/include   -O2 -pedantic -fomit-frame-pointer -fno-builtin" -c caml_z.c
    # ocamlmklib -failsafe -o zarith caml_z_arm.o caml_z.o -L/home/mirage/.opam/profile/lib/pkgconfig/../gmp-xen -lgmp-xen   -lgmp
    # ocamlopt   -c z.ml
    # ocamlopt   -c q.ml
    # ocamlopt   -c big_int_Z.ml
    # ocamlmklib -failsafe -o zarith z.cmx q.cmx big_int_Z.cmx -L/home/mirage/.opam/profile/lib/pkgconfig/../gmp-xen -lgmp-xen   -lgmp
    # ocamlopt -shared -o zarith.cmxs -I . zarith.cmxa
    ### stderr ###
    # [...]
    # caml_z.c: In function ‘ml_z_custom_hash’:
    # caml_z.c:2607:3: warning: ISO C90 forbids mixed declarations and code [-Wpedantic]
    #    mp_size_t i;
    #    ^
    # /usr/bin/ld: /home/mirage/.opam/profile/lib/pkgconfig/../gmp-xen/libgmp-xen.a(lt4-clear.o): relocation R_ARM_MOVW_ABS_NC against `__gmp_free_func' can not be used when making a shared object; recompile with -fPIC
    # /home/mirage/.opam/profile/lib/pkgconfig/../gmp-xen/libgmp-xen.a: error adding symbols: Bad value
    # collect2: error: ld returned 1 exit status
    # + cp libzarith.a /home/mirage/.opam/profile/lib/zarith/libzarith-xen.a
    # + mv /home/mirage/.opam/profile/lib/zarith/META /home/mirage/.opam/profile/lib/zarith/META.old
    # mv: cannot stat ‘/home/mirage/.opam/profile/lib/zarith/META’: No such file or directory

I then tried this:

    $ opam install zarith
    [NOTE] Package zarith is already installed (current version is 1.3).

This fixed it for me:

    $ opam reinstall zarith
    $ opam install zarith-xen

--- a/packages/zarith-xen/zarith-xen.1.3/opam
+++ b/packages/zarith-xen/zarith-xen.1.3/opam
@@ -5,8 +5,8 @@ build: [
   ["./mirage-install.sh"]
 ]
 remove: [
-  "rm" "-f" "%{prefix}%/lib/zarith/libzarith-xen.a"
-  "mv" "%{prefix}%/lib/zarith/META.old" "%{prefix}%/lib/zarith/META"
+  ["rm" "-f" "%{prefix}%/lib/zarith/libzarith-xen.a"]
+  ["mv" "%{prefix}%/lib/zarith/META.old" "%{prefix}%/lib/zarith/META"]
 ]
 depends: [
   "mirage-xen-posix"

